examples/platform/silabs/SiWx917/BUILD.gn

```diff
@@ -14,12 +14,49 @@
 
 import("//build_overrides/chip.gni")
 import("//build_overrides/efr32_sdk.gni")
+import("${chip_root}/examples/common/pigweed/pigweed_rpcs.gni")
 import("${chip_root}/src/lib/lib.gni")
 import("${chip_root}/src/platform/device.gni")
 import("${efr32_sdk_build_root}/SiWx917_sdk.gni")
 
 silabs_common_plat_dir = "${chip_root}/examples/platform/silabs"
 
+declare_args() {
+  enable_heap_monitoring = false
+
+  # OTA timeout in seconds
+  OTA_periodic_query_timeout = 86400
+
+  # Wifi related stuff - they are overridden by gn -args="use_wf200=true"
+  sl_wfx_config_softap = false
+  sl_wfx_config_scan = true
+
+  # Argument to Disable IPv4 for wifi(rs911)
+  chip_enable_wifi_ipv4 = false
+
+  # Argument to force enable WPA3 security on rs91x
+  rs91x_wpa3_only = false
+
+  #default WiFi SSID
+  chip_default_wifi_ssid = ""
+
+  #default Wifi Password
+  chip_default_wifi_psk = ""
+}
+
+silabs_common_plat_dir = "${chip_root}/examples/platform/silabs"
+
+import("${silabs_common_plat_dir}/SiWx917/args.gni")
+
+# Sanity check
+assert((chip_enable_wifi && use_SiWx917))
+if (chip_enable_wifi) {
+  assert(use_SiWx917)
+  import("${chip_root}/src/platform/silabs/SiWx917/wifi_args.gni")
+  import("SiWx917/rs911x.gni")
+}
```

Comment:
Yes You are right we don't require if condition here.
Asserts are required to check if the variables are set because we use these in build files which are common.

Reply:
Changed in Latest Commit